sqlx-macros-core/src/query/args.rs

```diff
@@ -114,11 +123,37 @@ pub fn quote_args<DB: DatabaseExt>(
     })
 }
 
-fn get_type_override(expr: &Expr) -> Option<&Type> {
+fn create_warning(name: Ident, ty: Type, expr: Expr) -> TokenStream {
+    let span = expr.span();
+    let stripped = strip_wildcard(expr).to_token_stream();
+    let current = quote!(#stripped: #ty).to_string();
+    let fix = quote!(#stripped as #ty).to_string();
+
+    let message = format!(
+        "
+\t\tType ascription pattern is deprecated, prefer casting
+\t\tTry changing from
+\t\t\t`{current}`
+\t\tto
+\t\t\t`{fix}`
+
+\t\tSee <https://github.com/rust-lang/rfcs/pull/3307> for more information
+"
+    );
+    let name = Ident::new(&format!("warning_{name}"), span);
```

Comment:
The names won't conflict as the `const`s are created inside the invocation's local scope.